### PR TITLE
docs: Add an introduction to actors

### LINKED ISSUE #648: Write a small introduction to actors

### PR DESCRIPTION
**Pull request by llucax**
This new document introduces the actor programming model and how to use it in Frequenz SDK.

Fixes #648.

### CODE REVIEW

**Comment by llucax** [comment]
Based on #667, ignore the first 4 commits.

File: docs/intro/actors.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        await input_sender.send("Hello")  # (6)!
+        msg = await output_receiver.receive()  # (7)!
+        print(msg)  # (10)!
+    # (11)!
+
+if __name__ == "__main__":  # (3)!
+    asyncio.run(main())
+```
+
+1. We define 2 actors: `Actor1` and `Actor2` that will just forward message
```
Comment (daniel-zullo-frequenz) [inline docs/intro/actors.md]:
```suggestion
1. We define 2 actors: `Actor1` and `Actor2` that will just forward a message
```

File: docs/intro/actors.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    * `middle_channel` is the channel that connects `Actor1` and `Actor2`.
+    * `output_channel` is the output channel for `Actor2`.
+
+5. We create two actors and use them as async context managers, `Actor1` and
+   `Actor2`, and connect them by creating new
+   [senders][frequenz.channels.Sender] and
+   [receivers][frequenz.channels.Receiver] from the channels.
+
+6. We schedule the [sending][frequenz.channels.Sender.send] of message `Hello`
+   to `Actor1` via `input_channel`.
```
Comment (daniel-zullo-frequenz) [inline docs/intro/actors.md]:
```suggestion
6. We schedule the [sending][frequenz.channels.Sender.send] of the message
   `Hello` to `Actor1` via `input_channel`.
```

**Review by daniel-zullo-frequenz — COMMENTED** [review]
I have a couple of cosmetic comments to check for. LGTM!

**Review by daniel-zullo-frequenz — DISMISSED** [review]

**Comment by llucax** [comment]
Did you review all the commits or only what wasn't in #667? If you reviewed everything maybe you can approve #667 too :D

**Review by daniel-zullo-frequenz — DISMISSED** [review]

**Review by daniel-zullo-frequenz — DISMISSED** [review]

**Review by daniel-zullo-frequenz — DISMISSED** [review]

**Comment by llucax** [comment]
Investigating an issue with numbered annotations rendering wrongly...

**Comment by llucax** [comment]
Seems to be an upstream bug: https://github.com/squidfunk/mkdocs-material/issues/6042, so I guess we can proceed hoping it will be fixed.

**Review by shsms — APPROVED** [review]
This looks good to me, but at some point, we should write additional sections on requirements, best practices, composition patterns, etc.

**Comment by llucax** [comment]
If you have any specific suggestions I can try to add some of that, it is weird, when I'm coding or reading issues I see a lot of things that it would be good to have in docs, but when I sit down to write docs I get a writers block and I don't know what to put in there :laughing:

**Comment by llucax** [comment]
But I think I will merge this since it is approved so we can keep moving, I can write up any suggestions you make later in a separate PR.

**Pull request merged**